scripts/huggingface.py

```diff
@@ -62,848 +88,1024 @@ def _get_accelerate_args(
-            if stop_sequences is None:
-                until = [self.eot_token]
-            else:
-                until = stop_sequences + [self.eot_token]
-
-            if max_generation_length is None:
-                max_tokens = self.max_gen_toks
+        elif "device_map" not in model_kwargs:
+            # set a device_map to initialize model on the right GPU.
+            # this is needed because it seems that the default behavior
+            # for quantized models now seems to be device_map="auto"
+            # which breaks data-parallel mode.
+            if hasattr(self, "accelerator"):
+                model_kwargs.update(
+                    {"device_map": {"": f"cuda:{self.accelerator.local_process_index}"}}
+                )
             else:
-                max_tokens = max_generation_length
-
-            token_context = self.tok_encode_batch(context)
-
-            responses = self._model_generate(     # pylint: disable=E1123, E1120
-                inputs=token_context,
-                max_tokens=max_tokens,
-                stop=until,
+                model_kwargs.update({"device_map": {"": str(self.device)}})
+
+        if not autogptq:
+            if model_kwargs.get("load_in_4bit", None):
+                assert (
+                    transformers.__version__ >= "4.30.0"
+                ), "load_in_4bit requires transformers >= 4.30.0"
+            if transformers.__version__ >= "4.30.0":
+                if model_kwargs.get("load_in_4bit", None):
+                    if model_kwargs.get("bnb_4bit_compute_dtype", None):
+                        model_kwargs["bnb_4bit_compute_dtype"] = get_dtype(
+                            model_kwargs["bnb_4bit_compute_dtype"]
+                        )
+            from intel_extension_for_transformers.transformers import (
+                AutoModelForCausalLM,
             )
-            responses = self.tok_decode(responses.tolist())
-
-            for response in responses:
-                # Ensure the generated responses do not contain the stop sequences.
-                for term in until:
-                    response = response.split(term)[0]
-                # partial caching
-                self.cache_hook.add_partial("greedy_until", (context, until), response)
-                results.append(response)
-        return reorder.get_original(results)
-
 
-class AutoCausalLM(HuggingFaceAutoLM):
-    """Causal language modeling.
-    You can find a set of supported models in the HF documentation:
-    https://huggingface.co/docs/transformers/main/model_doc/auto#transformers.AutoModelForCausalLM
-    """
-
-    AUTO_MODEL_CLASS = transformers.AutoModelForCausalLM
-    AUTO_PEFT_CLASS = peft.PeftModel
+            if self.model_format == "neural_speed":
+                from neural_speed import Model
+                self._model = Model()
+                self._model.init(pretrained, weight_dtype="int4", compute_dtype="int8",
```

Comment:
as mentioned above, we can get quant args from `kwargs`

Reply:
done